### DISCUSSION TITLE
PHP Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 268435456 bytes)

### DISCUSSION THREAD

**Discussion by burhancelebi** [laravel/framework#40443] [Q&A]
- Laravel Version: 8.75
- PHP Version: 8.1
- Database Driver & Version:

### Description:

I used Service and Repository design patterns. 
I created two Service Patterns called CartService and ProductVariantService and two Repositories called CartRepository and ProductVariantRepository.
I used CartService and ProductVariantRepository in ProductVariantService __construct method and i used CartRepository and ProductVariantService in CartService __construct method after that I got the error in the title. 
I changed memory_limit to -1 but it couldn't solve the my problem.
But when i removed CartService in ProductVariantService __construct method and when i removed ProductVariantService in CartService __construct method i could start the project.

Something like that :

```
class CartService
{
    /** @var CartRepository*/
    private CartRepository $cartRepository;

    /** @var ProductVariantsService */
    private ProductVariantsService $productVariantsService;

    public function __construct(CartRepository $cartRepository,
                                ProductVariantsService $productVariantsService)
    {
        $this->cartRepository = $cartRepository;
        $this->productVariantsService = $productVariantsService;
    }
}

class ProductVariantsService
{
    /**
     * @var ProductVariantsRepository
     */
    private ProductVariantsRepository $productVariantsRepository;

    /** @var CartService */
    private CartService $cartService;

    public function __construct(ProductVariantsRepository $productVariantsRepository,
                                CartService $cartService)
    {
        $this->productVariantsRepository = $productVariantsRepository;
        $this->cartService = $cartService;
    }
}
```

**Comment by koenhoeijmakers** [comment]
You've got recursive dependencies.

`ProductVariantsService` needs the `CartService` which needs a `ProductVariantsService` which needs a `CartService` and so on.

**Reply by burhancelebi** [reply]
Yes but is it should work or not ?
Actually I haven't tried any other non Laravel project.

**Reply by koenhoeijmakers** [reply]
Should not work, how would you resolve this?
